sockeye/training.py

```diff
@@ -586,7 +588,28 @@ def fit(self,
                 for name, val in metric_val.get_name_value():
                     logger.info('Checkpoint [%d]\tValidation-%s=%f', self.state.checkpoint, name, val)
 
-                # (3) update training metrics
+                # (2) determine stopping
```

Comment:
Why did this move before the actual update of `num_not_improved`? Wouldn't lead this to models training for one checkpoint longer than before?

Reply:
Reverted to (almost) the original step order and now manually adding validation metric to the loop that sets has_improved, since it wasn't yet available from self.state.metrics